editor/js/CmdAddObject.js

```diff
@@ -0,0 +1,105 @@
+/**
+ * @author dforrer / https://github.com/dforrer
+ * Developed as part of a project at University of Applied Sciences and Arts Northwestern Switzerland (www.fhnw.ch)
+ */
+
+/**
+ * @param object THREE.Object3D
+ * @constructor
+ */
+
+CmdAddObject = function ( object ) {
+
+	Cmd.call( this );
+
+	this.type = 'CmdAddObject';
+
+	this.object = object;
+	if ( object !== undefined ) {
+
+		this.name = 'Add Object: ' + object.name;
+
+	}
+
+};
+
+CmdAddObject.prototype = {
+
+	execute: function () {
+
+		this.editor.addObject( this.object );
+		this.editor.select( this.object );
+
+	},
+
+	undo: function () {
+
+		this.editor.removeObject( this.object );
+		this.editor.deselect();
+
+	},
+
+	toJSON: function () {
+
+		var output = Cmd.prototype.toJSON.call( this );
+
+		this.object.updateMatrixWorld( true );
+
```

Comment:
I am also looking into **CmdSetMaterial** and **CmdSetMaterialMap** for improvements in .toJSON() and .fromJSON().

But I am running into the problem that THREE.Material.toJSON() and THREE.Texture.toJSON() both require the presence of the *meta-object*, meaning they don't create it themselves if it doesn't exist.